xfrd-tcp.c

```diff
@@ -24,6 +24,82 @@
 #include "xfrd.h"
 #include "xfrd-disk.h"
 #include "util.h"
+#ifdef HAVE_TLS_1_3
+#include <openssl/ssl.h>
+#include <openssl/err.h>
+#endif
+
+#ifdef HAVE_TLS_1_3
+static SSL_CTX*
+create_ssl_context()
+{
+	SSL_CTX *ctx;
+	ctx = SSL_CTX_new(TLS_client_method());
+	if (!ctx) {
+		log_msg(LOG_ERR, "xfrd tls: Unable to create SSL ctxt");
+	}
+	else if (SSL_CTX_set_default_verify_paths(ctx) != 1) {
+		SSL_CTX_free(ctx);
+		log_msg(LOG_ERR, "xfrd tls: Unable to set default SSL verify paths");
+	}
+	/* Only trust 1.3 as per the specification */
+	else if (!SSL_CTX_set_min_proto_version(ctx, TLS1_3_VERSION)) {
+		SSL_CTX_free(ctx);
+		log_msg(LOG_ERR, "xfrd tls: Unable to set minimum TLS version 1.3");
+	}
```

Comment:
We should not return the deallocated ctx pointer, to stop double free errors.
```suggestion
	else if (!SSL_CTX_set_min_proto_version(ctx, TLS1_3_VERSION)) {
		SSL_CTX_free(ctx);
		log_msg(LOG_ERR, "xfrd tls: Unable to set minimum TLS version 1.3");
		return NULL;
	}
```

Reply:
Resolved in current code.